core/util/src/mill/util/CoursierConfig.scala

```diff
@@ -0,0 +1,53 @@
+package mill.util
+
+import coursier.CoursierEnv
+import coursier.cache.{CacheEnv, CachePolicy}
+import coursier.core.Repository
+import coursier.credentials.Credentials
+import coursier.params.Mirror
+
+import scala.concurrent.duration.Duration
+
+final case class CoursierConfig(
+    repositories: Seq[Repository],
+    confFiles: Seq[os.Path],
+    mirrors: Seq[Mirror],
+    cacheLocation: String,
+    archiveCacheLocation: String,
+    credentials: Seq[Credentials],
+    ttl: Option[Duration],
+    cachePolicies: Seq[CachePolicy]
+)
+
+object CoursierConfig {
+  // as much as possible, prefer using CoursierConfigModule.coursierConfig(),
+  // that reads up-to-date values from the environment and from Java properties
+  def default(): CoursierConfig =
```

Comment:
In which situation do we need this directly? Maybe, making this `private[mill]` is an option?